src/miner.cpp

```diff
@@ -526,29 +519,47 @@ void BlockAssembler::addPackageTxs(int& nPackagesSelected, int& nDescendantsUpda
             continue;
         }
 
-        // This transaction will make it in; reset the failed counter.
-        nConsecutiveFailed = 0;
-
         // Package can be added. Sort the entries in a valid order.
         std::vector<CTxMemPool::txiter> sortedEntries;
         SortForBlock(ancestors, sortedEntries);
 
+        // Test pocketnet part for all ancestors
+        bool testPocketnetPart = true;
+        for (CTxMemPool::txiter it : sortedEntries)
+        {
+            if (!TestTransaction(it->GetSharedTx()))
```

Comment:
@tawmaz I am confused by the validation logic in my pool here. That if the bundle of transactions is not accepted in full, then pocket Black will be filled with a part of the checked bundle and will remain there. I think another solution is needed here..

Reply:
Do you want to revert this commit from the release for now?

Reply:
No, I plan to rewrite this logic tomorrow so that TestTransaction adds the checked pocketnet part and inserts it into a temporary array, which after the successful completion of the cycle will be inserted into the final pocketBlock.